### PR TITLE
Svls 6036 set timeouts

### PR DESCRIPTION
**Pull request by alexgallotta**
Allow passing a timeout to the client

### CODE REVIEW

**Review by duncanista — APPROVED** [review]

File: serverless/src/main.rs [hunk 1]
```diff
@@ -160,6 +160,7 @@ async fn start_dogstatsd(
                 Arc::clone(&metrics_aggr),
                 build_fqdn_metrics(dd_site),
                 https_proxy,
+                Duration::from_secs(5),
```
Comment (ekump) [inline serverless/src/main.rs:163]: nitpick: I typically like to put these sorts of values in consts so it's clearer what their purpose is.

File: dogstatsd/src/datadog.rs [hunk 2]
```diff
@@ -20,8 +21,13 @@ pub struct DdApi {
 
 impl DdApi {
     #[must_use]
-    pub fn new(api_key: String, site: String, https_proxy: Option<String>) -> Self {
-        let client = match Self::build_client(https_proxy) {
+    pub fn new(
```
Comment (ekump) [inline dogstatsd/src/datadog.rs:24]: nitpick: I think it's ok right now, but if we go beyond four parameters here we should consider implementing a Dogstatsd builder.

**Review by ekump — APPROVED** [review]

File: serverless/src/main.rs (hunk 1 above)
Comment (duncanista) [reply serverless/src/main.rs:163]: On it

File: dogstatsd/src/datadog.rs (hunk 2 above)
Comment (duncanista) [reply dogstatsd/src/datadog.rs:24]: Agree, I'll do it in another PR tho!

**Pull request merged**